oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/ElasticResponseListener.java

```diff
@@ -68,8 +68,9 @@ default void startData(long totalHits) { /*empty*/ }
         /**
          * This method is called for each {@link Hit} retrieved
          * @param searchHit a search result
+         * @return true if the search hit was successfully processed, false otherwise
          */
-        void on(Hit<ObjectNode> searchHit);
```

Comment:
This change atm might be considered a breaking change imo. 
If we know for sure that this interface is not implemented outside oak, we should probably mark it with @ProviderType.